.github/workflows/android-master.yml

```diff
@@ -0,0 +1,58 @@
+# A workflow for controlling the acions taken by CI whenever we merge code into master
+
+name: Android SDK Master CI
+
+# Triggers the workflow on when code is merged into master
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  # This workflow contains a single job for running our tests
+  test:
+    name: Run Unit Tests
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so the job can access it
+      - uses: actions/checkout@v1
+
+      # Sets up a JDK on the job so that we can run Java code
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      # Call Gradle to run the unit tests
+      - name: Unit Tests
+        run: bash ./gradlew test --stacktrace
+
+    aar:
+      name: Generate AAR
+      # The type of runner that the job will run on
+      runs-on: ubuntu-20.04
+
+      steps:
+        # Checks-out your repository under $GITHUB_WORKSPACE, so the job can access it
+        - uses: actions/checkout@v1
+
+        # Sets up a JDK on the job so that we can run Java code
+        - name: Set up JDK 1.8
+          uses: actions/setup-java@v1
+          with:
+              java-version: 1.8
+
+        # Call Gradle to build our our debug AAR
+        - name: Build debug AAR
+          run: bash ./gradlew assembleDebug --stacktrace
+
+        # Stores the generated AAR to an outputs folder
+        - name: Store AAR
+          uses: actions/upload-artifact@v1
```

Comment:
we'll want to use v2 of the action https://github.com/actions/upload-artifact note here about version syntax: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-versioned-actions

Reply:
also one thing to note about artifacts, they will automatically expire and be deleted after 90 days

Reply:
That's ok with me I think. The storing of artifacts currently is a precursor step to us sending them off to Firebase Test Lab when we have that infrastructure set up. I don't think we'll need old SDKs for that.